### ISSUE TITLE
Full screen and detail page

### ISSUE THREAD

**Issue by Aragunz** [MillmanY/MMPlayerView#10]
Hi there dear sir,
I have an app which is similar to utube.
I want player to play after you present a viewcontroller, and when i rotate full screen, first time it works, after that the full screen doesnt work when i close the presented viewcontroller and open another one, doesnt update the latest video.
How do i approach this?

Here is the code i use on detail page:

1. This is how i present the player container controller.
```
 let playerController = storyboard?.instantiateViewController(withIdentifier: "GVPlayerController") as! GVPlayerController
        playerController.videoObjectBasic = videoBasic
       //self.navigationController?.pushViewController(playerController, animated: false)
        self.present(playerController, animated: true)
```
2. This is the mmplayer layer, and playerContainer which is a uiview where the player will be shown.

 ```
 var videoObjectBasic: GVVideoBasic?

    lazy var mmPlayerLayer: MMPlayerLayer = {
        let l = MMPlayerLayer()
        l.cacheType = .memory(count: 5)
        l.coverFitType = .fitToPlayerView
        l.videoGravity = AVLayerVideoGravity.resizeAspect
        l.replace(cover: GVPlayerView.instantiateFromNib())
        return l
    }()

    @IBOutlet var playerContainer: UIView!
```
3. Here im adding the observer for container full screen

 ```
 override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        NotificationCenter.default.addObserver(forName: .UIDeviceOrientationDidChange, object: nil, queue: nil) { [unowned self] (_) in
            self.landscapeAction()
        }

    }

fileprivate func landscapeAction() {

        if UIDevice.current.orientation.isLandscape {
            let full = self.storyboard?.instantiateViewController(withIdentifier: "GVFullScreenPlayerController") as! GVFullScreenPlayerController
            MMLandscapeWindow.shared.makeKey(root: full, playLayer: mmPlayerLayer, completed: {
                print("landscape completed")
                //                    self.playerCollect.isScrollEnabled = true
            })
        }
    }
```
4. And lastly here is the player i load with a link from API
private func loadPlayer(urlString:String) {

 ```
       if !MMLandscapeWindow.shared.isKeyWindow {
            self.mmPlayerLayer.playView = self.playerContainer
        }

        self.mmPlayerLayer.set(url: URL.init(string:urlString)!, state: { (status) in
            switch status {
            case .failed(let err):
                let alert = UIAlertController(title: "err", message: err.description, preferredStyle: .alert)
                alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
                self.present(alert, animated: true, completion: nil)
            case .ready:
                print("Ready to Play")
            case .playing:
                print("Playing")
            case .pause:
                print("Pause")
            case .end:
                print("End")
            default: break
            }
        })
        self.mmPlayerLayer.startLoading()
    }
```
Basically it doesn't change or load, second video when first video is dismissed.
What im doing wrong ?

Thank you for your time

**Comment by Aragunz** [comment]
Commenting  `if self.isKeyWindow` fixes the problem, but is this right ?
```
 public func makeKey(root: UIViewController, playLayer: MMPlayerLayer, completed: (()-> Void)?) {
        
//        if self.isKeyWindow {
//            return
//        }
        currentPlayLayer = playLayer
        originalPlayView = playLayer.playView
        self.rootViewController = root
        self.currentPlayLayer?.clearURLWhenChangeView = false
        currentPlayLayer?.playView = root.view
        self.completed = completed
        self.makeKeyAndVisible()
    }
```

**Comment by MillmanY** [comment]
How about like this ?
```
class GVPlayerController: UIViewController {
    var videoObjectBasic: GVVideoBasic?
    var observer: Any?

    lazy var mmPlayerLayer: MMPlayerLayer = {
        let l = MMPlayerLayer()
        l.cacheType = .memory(count: 5)
        l.coverFitType = .fitToPlayerView
        l.videoGravity = AVLayerVideoGravity.resizeAspect
        l.replace(cover: GVPlayerView.instantiateFromNib())
        return l
    }()
    
    @IBOutlet var playerContainer: UIView!

    override func viewDidLoad() {
        super.viewDidLoad()
        mmPlayerLayer.playView = playerContainer
        
        self.mmPlayerLayer.set(url: URL.init(string:urlString)!, state: { (status) in
            switch status {
            case .failed(let err):
                let alert = UIAlertController(title: "err", message: err.description, preferredStyle: .alert)
                alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
                self.present(alert, animated: true, completion: nil)
            case .ready:
                print("Ready to Play")
            case .playing:
                print("Playing")
            case .pause:
                print("Pause")
            case .end:
                print("End")
            default: break
            }
        })
        self.mmPlayerLayer.startLoading()
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        observer = NotificationCenter.default.addObserver(forName: .UIDeviceOrientationDidChange, object: nil, queue: nil) { [unowned self] (_) in
            self.landscapeAction()
        }
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        if let o = observer {
            NotificationCenter.default.removeObserver(o)
        }
        observer = nil
    }
    
    fileprivate func landscapeAction() {
        if UIDevice.current.orientation.isLandscape  {
            return
        }
        
           let full = self.storyboard?.instantiateViewController(withIdentifier: "GVFullScreenPlayerController") as! GVFullScreenPlayerController
        
        MMLandscapeWindow.shared.makeKey(root: full, playLayer: self.playLayer, completed: { [weak self] in


        })
    }
}
```

**Comment by Aragunz** [comment]
@MillmanY It worked like a charm, if u have few seconds to explain it technically prbably ppl in future will need this. Thank you :).

**Comment by MillmanY** [comment]
Ok, I will

If you need more information, jsut let me know

**Comment by rossman22590** [comment]
Type 'MMLandscapeWindow' has no member 'shared' please help

**Comment by rossman22590** [comment]
my code:  fileprivate func landscapeAction() {
     
        if UIDevice.current.orientation.isLandscape {
             let full = UIStoryboard.init(name: "LoggedInTabbar", bundle: nil).instantiateViewController(withIdentifier: "FullScreenVC") as! FullScreenVC
            MMLandscapeWindow.shared.makeKey(root: full, playLayer: self.mmPlayerLayer, completed: {
                print("landscape completed")
            })
        }
    }
